Reference/google/protobuf/any_test.pb.swift

```diff
@@ -123,35 +108,42 @@ struct ProtobufUnittest_TestAny: SwiftProtobuf.Proto3Message, SwiftProtobuf._Mes
   init() {}
 
   mutating func _protoc_generated_decodeMessage<D: SwiftProtobuf.Decoder>(decoder: inout D) throws {
-    try _uniqueStorage().decodeMessage(decoder: &decoder)
+    try withExtendedLifetime(_storage) { (_storage: _StorageClass) in
+      while let fieldNumber = try decoder.nextFieldNumber() {
+        try decodeField(decoder: &decoder, fieldNumber: fieldNumber)
+      }
+    }
   }
 
   mutating func _protoc_generated_decodeField<D: SwiftProtobuf.Decoder>(decoder: inout D, fieldNumber: Int) throws {
-    try _uniqueStorage().decodeField(decoder: &decoder, fieldNumber: fieldNumber)
+    switch fieldNumber {
```

Comment:
We might be able to improve this some by breaking up `decodeField` into a static func that takes the storage reference it's given.  That way, decodeMessage still only pays CoW overhead once, not once per field:
```
mutating func _protoc_generated_decodeMessage(...) {
   _ensureUniqueStorage()
   withExtendedLifetime ... {
      while let fieldNumber = nextFieldNumber {
          MyType._protobuf_generated_static_decodeField(_storage, ...)
      }
   }
}

mutating func _protobuf_generated_decodeField(...) {
    _ensureUniqueStorage()
    withExtendedLifetime {
         MyType._protobuf_generated_static_decodeField(storage: _storage, etc...)
    }
}

private static func _protobug_generated_static_decodeField(storage: _StorageClass, etc...) {
     ... do decodeField work without CoW here...
}
```

The extra hit for groups is unfortunate but I haven't yet come up with a clean way to avoid it.